viewing_party/party.py

```diff
@@ -1,23 +1,122 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    new_movie = {}
+    if not title or not genre or not rating:
+        return None
+    else:
+        new_movie["title"] = title
+        new_movie["genre"] = genre
+        new_movie["rating"] = rating
+    return new_movie
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    for i in range(len(user_data["watchlist"])):
+        if (user_data["watchlist"][i]["title"]) == title:
+            watched_movie = (user_data["watchlist"].pop(i))
+            user_data["watched"].append(watched_movie)
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    sum_of_ratings = 0
+    if not user_data["watched"]:
+        return 0.0
+    else:
+        for movie in user_data["watched"]:
+            sum_of_ratings += movie["rating"]
+        average_rating = sum_of_ratings/len(user_data["watched"])
+    return average_rating
+
+def get_most_watched_genre(user_data):
+    genre_counter = {}
+    if not user_data["watched"]:
+        return None
+    else:
+        for movie in user_data["watched"]:
+            if movie["genre"] not in genre_counter:
+                genre_counter[movie["genre"]] = 1
+            elif movie["genre"] in genre_counter:
+                genre_counter[movie["genre"]] += 1
+        most_frequent_genre = max(genre_counter, key=genre_counter.get)

```

Comment:
Nice job using a Python built-in that we haven't discussed in class! In the future, when pulling in code from beyond what we've talked about, it's great if you could include a comment about the research that you did, and how you think it works. And it's still worth implementing similar functionality a few times yourself, as it can help practice building our fundamental skills. For now, our goal should be getting as much coding practice as possible!